python/gym_jiminy/toolbox/gym_jiminy/toolbox/math/generic.py

```diff
@@ -59,18 +59,128 @@ def quat_to_rpy(quat: np.ndarray) -> np.ndarray:
     """Compute the Yaw-Pitch-Roll Euler angles representation of a single or a
     batch of quaternions.
 
-    :param quat: 1D array or a 2D array whose first dimension corresponds to
-                 the number of individual quaternions, and the second to the 4
-                 coordinates [qx, qy, qz, qw].
+    :param quat: 1D array, a 2D array or nD array whose first dimensions
+                 corresponds to the number of individual quaternions, and the
+                 last to the 4 coordinates [qx, qy, qz, qw].
     """
-    qx, qy, qz, qw = np.atleast_2d(quat)[:, -4:].T
+    qx, qy, qz, qw = np.atleast_2d(quat)[..., -4:].T
```

Comment:
Is transpose really enough for N-d arrays in the general case (more than 3-d typically) ?